tutorials/settings_tutorial.md

```diff
@@ -0,0 +1,129 @@
+The remainder of the class is simple accessors and mutators to encapsulate the UI components used on the `JPanel`.
+
+
+## The AppSettingsConfigurable Class
+The role of [`AppSettingsConfigurable`](https://github.com/JetBrains/intellij-sdk-docs/tree/master/code_samples/settings/src/main/java/org/intellij/sdk/settings/AppSettingsConfigurable.java) is analogous to a Controller in MVC, with `AppSettingsComponent` being the View, and `AppSettingsState` acting as the Model.
+Consequently, the methods of `AppSettingsConfigurable` are called by the IntelliJ Platform, and `AppSettingsConfigurable` in turn interacts with `AppSettingsComponent` and `AppSettingState`.
+
+### Declaring the AppSettingsConfigurable Using an EP
+As described in [Declaring Application Settings](/reference_guide/settings_guide.md#ep-for-declaring-application-settings), the `com.intellij.applicationConfigurable` is used as the EP. 
+Detailed definitions of these attributes can be found in [Settings EP Attributes](/reference_guide/settings_guide.md#settings-ep-attributes): 
+
+```xml
+  <extensions defaultExtensionNs="com.intellij">
+    <applicationConfigurable parentId="tools" instance="org.intellij.sdk.settings.AppSettingsConfigurable"
+                             id="org.intellij.sdk.settings.AppSettingsConfigurable"
+                             displayName="SDK: Application Settings Example"/>
+  </extensions>
+```
+
+
+### Creating the AppSettingsConfigurable Implementation
+The `AppSettingsConfigurable` class implements [`Configurable`](upsource:///platform/platform-api/src/com/intellij/openapi/options/Configurable.java) interface.
+The class has one field to hold a reference to the `AppSettingsComponent`, once it is created.
+As [required](/reference_guide/settings_guide.md#constructors) there is a default constructor with no arguments, and for performance reasons the constructor creates no UI elements.
+
+```java
+{% include /code_samples/settings/src/main/java/org/intellij/sdk/settings/AppSettingsConfigurable.java %}
+```
+
+#### AppSettingsConfigurable Methods
+All the methods in this class are overrides of the methods in the `Configurable` interface. 
+Readers are encouraged to review the Javadoc comments for the `Configurable` methods.
+There are also [Settings Guide notes](/reference_guide/settings_guide.md#intellij-platform-interactions-with-configurable) about when `Configurable` methods are called by the IntelliJ Platform.
+
+## Testing the Custom Settings Plugin
+After performing the steps described above, compile and run the plugin in a Development Instance to see the custom Settings available in the Settings Dialog.
+Open the IDE Settings by selecting **Settings/Preferences \| Tools \| SDK: Application Settings Example**.
+The settings are preloaded with the default values:
+
+!["Settings Defaults"](img/settings_defaults.png){:width="800px"}
```

Comment:
if we want these screenshots, please redo them (after fixing layout) with less vertical space